boards: stm32: Add nucleo_h533re support

marwaiehm-st opened the pull request:
Add DTS and SoC support for STM32h533Xe familly
Add board support for STM32H533RE

marwaiehm-st committed:
dts: arm: st: add stm32h533Xe support

Provide support for STM32H533XE family support

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

marwaiehm-st committed:
soc: st: stm32: add stm32h533xx support

Add STM32H533XX familly support

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

marwaiehm-st committed:
boards: st: add stm32h533re board support

Add board support for stm32h533re

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

marwaiehm-st committed:
dts: arm: st: h5: Add node gpiof

Add the node gpiof to the stm32h5 dtsi file.

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

erwango commented on dts/arm/st/h5/stm32h533Xe.dtsi:
```diff
@@ -0,0 +1,35 @@
+	soc {
+		flash-controller@40022000 {
+			flash0: flash@8000000 {
+				reg = <0x08000000 DT_SIZE_K(215)>;
```
```suggestion
				reg = <0x08000000 DT_SIZE_K(512)>;
```
?

erwango reviewed (changes requested):
Otherwise LGTM

ajarmouni-st commented on boards/st/nucleo_h533re/nucleo_h533re.dts:
```diff
@@ -0,0 +1,129 @@
+	model = "STMicroelectronics STM32H533RE-NUCLEO board";
+	compatible = "st,stm32h533re-nucleo";
+
+	#address-cells = <1>;
+	#size-cells = <1>;
```
Is this needed?

ajarmouni-st commented on boards/st/nucleo_h533re/nucleo_h533re.dts:
```diff
@@ -0,0 +1,129 @@
+
+	leds: leds {
+		compatible = "gpio-leds";
+		green_led_2: led_42{
```
nit
```suggestion
		green_led_2: led_42 {
```

ajarmouni-st commented on boards/st/nucleo_h533re/nucleo_h533re.yaml:
```diff
@@ -0,0 +1,17 @@
+  - gpio
+  - watchdog
+  - pwm
+  - dma
```
```
- rtc
```

ajarmouni-st commented on boards/st/nucleo_h533re/nucleo_h533re.dts:
```diff
@@ -0,0 +1,129 @@
+	clocks = <&rcc STM32_CLOCK_BUS_APB3 0x00200000>,
+		 <&rcc STM32_SRC_LSE RTC_SEL(1)>;
+	status = "okay";
+};
```
Could you add an overlay for this board in `tests/drivers/rtc/rtc_api/boards` as well?

ajarmouni-st commented on boards/st/nucleo_h533re/nucleo_h533re.dts:
```diff
@@ -0,0 +1,129 @@
+	status = "okay";
+};
+
+&gpdma1 {
+	status = "okay";
+};
+
+&gpdma2 {
+	status = "okay";
+};
```
Do these need to be enabled?

ajarmouni-st commented on boards/st/nucleo_h533re/nucleo_h533re.yaml:
```diff
@@ -0,0 +1,17 @@
+  - gpio
+  - watchdog
+  - pwm
+  - dma
```
have you tested `dma`? also for this tag to be included I think an overlay should be added to `tests/drivers/dma/loop_transfer/boards`

marwaiehm-st replied:
No i didn't tested the dma, i think should be removed from this PR and treated on sperated one.

ajarmouni-st commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,317 @@
+.. _nucleo_h533re_board:
+
+ST Nucleo H533re
```
for consistency across doc
```suggestion
ST Nucleo H533RE
```

ajarmouni-st commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,317 @@
+  - Virtual COM port
+  - debug port
+
+- One user LED shared with ARDUINO® Uno V3
```
```suggestion
- One user LED shared with ARDUINO |reg| Uno V3
```

ajarmouni-st commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,317 @@
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+| DAC       | on-chip    | DAC Controller                      |
```
Not enabled in DTS

ajarmouni-st commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,317 @@
++-----------+------------+-------------------------------------+
+| RTC       | on-chip    | Real Time Clock                     |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi bus                             |
```
Not enabled in DTS

ajarmouni-st commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,317 @@
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi bus                             |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c bus                             |
```
Not enabled in DTS

marwaiehm-st committed:
tests: drivers: rtc: rtc_api: Add nucleo_h533re

Adds necessary overlay nucleo_h533re in rtc_api test case
to enables the board.

Signed-off-by: IBEN EL HADJ MESSAOUD Marwa <marwa.ibenelhadjmessaoud-ext@st.com>

marwaiehm-st commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,317 @@
+Debugging
+=========
+
+You can debug an application in the usual way.  Here is an example for the
```
No, PyOCD is supported

FRASTM commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,306 @@
+
+- One user LED shared with ARDUINO |reg| Uno V3
+- Two push-buttons: USER and RESET
+- 32.789 kHz crystal oscillator
```
```suggestion
- 32.768 kHz crystal oscillator
```

FRASTM commented on boards/st/nucleo_h533re/doc/index.rst:
```diff
@@ -0,0 +1,306 @@
+
+Nucleo H533RE System Clock could be driven by internal or external oscillator,
+as well as main PLL clock. By default System clock is driven by PLL clock at
+240MHz, driven by 8MHz external clock provided from the STLINK-V3EC.
```
There is a 24MHz (X3) for the HSE external clock on the nucleo_h533 board
--> HSE is not STLINK sourced

FRASTM commented on boards/st/nucleo_h533re/nucleo_h533re.dts:
```diff
@@ -0,0 +1,122 @@
+&rcc {
+	clocks = <&clk_hsi>;
+	clock-frequency = <DT_FREQ_M(64)>;
+	ahb-prescaler = <1>;
```
we Could also increase the sysclock with the  PLL

FRASTM commented on soc/st/stm32/stm32h5x/Kconfig.defconfig.stm32h533xx:
```diff
@@ -0,0 +1,11 @@
+if SOC_STM32H533XX
+
+config NUM_IRQS
+	default 134
```
can you check, RM0481 gives only positions 0 to  132 for the  STM32H523/533xx vector table

FRASTM reviewed (commented):
LGTM
